swift/src/IceImpl/DispatchAdapter.mm

```diff
@@ -16,31 +16,37 @@
     const std::byte* inEncaps;
     std::function<void()> cleanup;
 
-    // An InputSteam can contain one or more requests when we're processing a batch request. In the case when there
-    // is more than one request in the the batch we need to copy the encapsulation from the InputStream to a
-    // heap allocated vector as the remainder of the memory is needed for subsequent requests.
-    if (request.requestCount() > 1)
+    // The Swift side can asynchronously unmarshal the request, so we need to either move or copy the encapsulation
+    // data to ensure it remains alive until Swift completes unmarshaling. The data should then be freed once the
```

Comment:
I would remove the second sentence. Obviously we don't leak this memory. It's also not ideal to keep it around until the dispatch finishes; ideally, we would release it immediately after unmarshaling the incoming args.

Reply:
> we would release it immediately after unmarshaling the incoming args.

This would break collocation retries. If the dispatch throws a exception that can be retried.